logstash-core/lib/logstash/environment.rb

```diff
@@ -34,6 +34,7 @@ module Environment
   end
 
   [
+            Setting::String.new("on_superuser", "WARN"),
```

Comment:
Since we'll want to guide users towards not using root, instead of these three options we could instead do:
1. (default) allow but log deprecation warn
2. block

With only two options we can instead have a boolean setting:

```suggestion
            Setting::Boolean.new("allow_superuser", true),
```

Reply:
Thanks @jsvd for the suggestion.
Initially had a same thought. We with @yaauie had a discussion on this and we concluded that option should be `open to extend` in the future. You can take a look at his comments under this PR and let me know if you have still concern.
Thanks.